### PR TITLE
Updated DOM and local storage for survey answers.

### PR DESCRIPTION
**Pull request by arkassay**
Will add styles and analytics in next PR.
Updates #3.

### CODE REVIEW

File: google-codelab-survey/BUILD [hunk 1]
```diff
@@ -11,6 +12,7 @@ filegroup(
         "*.html",
         "*.png",
     ]) + [
+        "google_codelab_survey_scss_bin",
```
Comment (skarEE) [inline google-codelab-survey/BUILD]: :google_codelab_survey_scss_bin

File: google-codelab-survey/google_codelab_survey.js [hunk 2]
```diff
@@ -33,33 +66,118 @@ class CodelabSurvey extends HTMLElement {
...
    * @export
    * @override
    */
   connectedCallback() {
+    this.checkStoredData_();
     this.updateDom_();
+    this.bindEvents_();
+  }
+
+  bindEvents_() {
```
Comment (skarEE) [inline google-codelab-survey/google_codelab_survey.js]: functions need 
@privates

File: google-codelab-survey/google_codelab_survey.js [hunk 3]
```diff
@@ -33,33 +66,118 @@ class CodelabSurvey extends HTMLElement {
...
+    const surveyQuestions = document.querySelectorAll(
+      Selectors.OPTIONS_WRAPPER);
+    surveyQuestions.forEach(el => {
+      this.eventHandler_.listen(el, events.EventType.CLICK, (e) => {
+        this.handleSurveyClick_(e.currentTarget);
+      });
+    });
+  }
+
+  handleSurveyClick_(surveyQuestionEl) {
```
Comment (skarEE) [inline google-codelab-survey/google_codelab_survey.js]: jsdoc param

File: google-codelab-survey/google_codelab_survey.js [hunk 4]
```diff
@@ -33,33 +66,118 @@ class CodelabSurvey extends HTMLElement {
...
   connectedCallback() {
+    this.checkStoredData_();
     this.updateDom_();
+    this.bindEvents_();
+  }
+
+  bindEvents_() {
+    const surveyQuestions = document.querySelectorAll(
+      Selectors.OPTIONS_WRAPPER);
+    surveyQuestions.forEach(el => {
```
Comment (skarEE) [inline google-codelab-survey/google_codelab_survey.js]: What about a single event listener on the parent?

File: google-codelab-survey/google_codelab_survey.js [hunk 5]
```diff
@@ -33,33 +66,118 @@ class CodelabSurvey extends HTMLElement {
...
+    this.setAnsweredQuestions_();
+    this.setAttribute(SURVEY_UPGRADED_ATTR, '');
+  }
+
+  setAnsweredQuestions_() {
+    const surveyData = this.storedData_[this.surveyName_];
+    if (surveyData) {
+      Object.keys(surveyData).forEach(key => {
+        const id = this.normalizeIdAttr_(surveyData[key]);
+        const inp = this.querySelector(`[id="${id}"]`);
```
Comment (x1ddos) [inline google-codelab-survey/google_codelab_survey.js]: wouldn't a simpler query work as well? `querySelector(#${id})`

File: .gitignore [hunk 6]
```diff
@@ -1,3 +1,3 @@
 bazel-*
-.*
+.DS_Store
 !/.gitignore
```
Comment (x1ddos) [inline .gitignore]: But now `!/.gitignore` isn't needed.

**Comment by x1ddos** [comment]
The `testCodelabSurveyUpgraded` test now fails on the CI with:

```
[tools/webtest] 2018/05/03 20:29:09 GET /google-codelab-survey/google_codelab_survey_test_bin.js
[tools/webtest] 2018/05/03 20:29:09 testCodelabSurveyIncorrectFormatNotUpgraded: OK
[tools/webtest] 2018/05/03 20:29:09 testCodelabSurveyLoadsStoredAnswers: OK
[tools/webtest] 2018/05/03 20:29:09 testCodelabSurveyOptionClick: OK
[tools/webtest] 2018/05/03 20:29:09 1 out of 4 test(s) failed:
testCodelabSurveyUpgraded: FAIL
1. Expected not to be <null>
Error: Expected not to be <null>
    at $goog$testing$asserts$raiseException$$ (http://127.0.0.1:39030/google-codelab-survey/google_codelab_survey_test_bin.js:2872:37)
    at $_assert$$ (http://127.0.0.1:39030/google-codelab-survey/google_codelab_survey_test_bin.js:2704:22)
    at $$goog$testing$asserts$assertNotNull$$$ (http://127.0.0.1:39030/google-codelab-survey/google_codelab_survey_test_bin.js:2757:3)
    at Object.testCodelabSurveyUpgraded (http://127.0.0.1:39030/google-codelab-survey/google_codelab_survey_test_bin.js:3945:3)
    at $goog$testing$TestCase$$.$$JSCompiler_prototypeAlias$$$$invokeFunction_$$ [as $invokeFunction_$] (http://127.0.0.1:39030/google-codelab-survey/google_codelab_survey_test_bin.js:3244:46)
    at $goog$testing$TestCase$$.$$JSCompiler_prototypeAlias$$$$safeRunTest_$$ [as $safeRunTest_$] (http://127.0.0.1:39030/google-codelab-survey/google_codelab_survey_test_bin.js:3231:15)
    at $goog$testing$Continuation_$run$$ (http://127.0.0.1:39030/google-codelab-survey/google_codelab_survey_test_bin.js:3583:68)
    at $goog$testing$TestCase$$.<anonymous> (http://127.0.0.1:39030/google-codelab-survey/google_codelab_survey_test_bin.js:3200:191)
    at $goog$testing$Continuation_$run$$ (http://127.0.0.1:39030/google-codelab-survey/google_codelab_survey_test_bin.js:3583:68)
    at $goog$testing$TestCase$$.<anonymous> (http://127.0.0.1:39030/google-codelab-survey/google_codelab_survey_test_bin.js:3199:5)
    at new $goog$Promise$$ (http://127.0.0.1:39030/google-codelab-survey/google_codelab_survey_test_bin.js:1969:28)
    at $goog$testing$TestCase$$.$$JSCompiler_prototypeAlias$$$$runTestsReturningPromise$$ [as $runTestsReturningPromise$] (http://127.0.0.1:39030/google-codelab-survey/google_codelab_survey_test_bin.js:3198:3)
    at $goog$testing$TestRunner$$.$$JSCompiler_prototypeAlias$$$$execute$$ [as $execute$] (http://127.0.0.1:39030/google-codelab-survey/google_codelab_survey_test_bin.js:3658:84)
    at http://127.0.0.1:39030/google-codelab-survey/google_codelab_survey_test_bin.js:3782:24
2018/05/03 20:29:09 test failed exit status 1
```

Do they pass locally?

File: .gitignore [hunk 7]
```diff
@@ -1 +1,3 @@
 bazel-*
+.DS_Store
+!/.gitignore
```
[the remark this answers is not in the document]
Comment (arkassay) [reply .gitignore]: woops! done :)

**Comment by arkassay** [comment]
Fixed the tests this should be ready for your review again!

File: .gitignore [hunk 8]
```diff
@@ -1 +1,3 @@
 bazel-*
+.*
+!/.gitignore
```
[the remark this answers is not in the document]
Comment (arkassay) [reply .gitignore]: Done!

File: google-codelab-survey/google_codelab_survey.js (hunk 2 above)
Comment (arkassay) [reply google-codelab-survey/google_codelab_survey.js]: done

File: google-codelab-survey/google_codelab_survey.js (hunk 4 above)
Comment (arkassay) [reply google-codelab-survey/google_codelab_survey.js]: Done!

File: google-codelab-survey/google_codelab_survey.js (hunk 3 above)
Comment (arkassay) [reply google-codelab-survey/google_codelab_survey.js]: Done.

File: google-codelab-survey/google_codelab_survey.js [hunk 9]
```diff
@@ -33,33 +68,144 @@ class CodelabSurvey extends HTMLElement {
...
+   * @private
+   */
+  handleClick_(el) {
+    const isOptionWrapper = el.classList.contains(
+      CssClass.RADIO_WRAPPER);
+    const elParent = /** @type {!Element} */ (el.parentElement);
+    const isOptionChild = elParent.classList.contains(CssClass.RADIO_WRAPPER);
+    if (isOptionWrapper || isOptionChild) {
+      let optionEl = el;
+      if (isOptionChild) optionEl = elParent;
```
Comment (skarEE) [inline google-codelab-survey/google_codelab_survey.js]: add parens

File: google-codelab-survey/google_codelab_survey.js [hunk 10]
```diff
@@ -33,33 +68,144 @@ class CodelabSurvey extends HTMLElement {
...
    */
   connectedCallback() {
+    this.checkStoredData_();
     this.updateDom_();
+    this.bindEvents_();
+  }
+
+  /** @private */
+  bindEvents_() {
+    this.eventHandler_.listen(document.body, events.EventType.CLICK, (e) => {
```
Comment (skarEE) [inline google-codelab-survey/google_codelab_survey.js]: You can also do:
this.eventHandler_.listen(document.body, events.EventType.CLICK,
    (e) => this.handleClick_(e.target));

File: google-codelab-survey/google_codelab_survey.js [hunk 11]
```diff
@@ -33,33 +68,144 @@ class CodelabSurvey extends HTMLElement {
...
+      this.handleOptionSelected_(optionEl);
+    }
   }
 
+  /**
+   * @param {!Element} optionEl
+   * @private
+   */
+  handleOptionSelected_(optionEl) {
+    const optionTextEl = /** @type {!Element} */ (
```
Comment (skarEE) [inline google-codelab-survey/google_codelab_survey.js]: Is this guaranteed to always be not null?

File: google-codelab-survey/google_codelab_survey.js [hunk 12]
```diff
@@ -33,33 +68,144 @@ class CodelabSurvey extends HTMLElement {
...
+  }
+
+  /** @private */
+  setAnsweredQuestions_() {
+    const surveyData = this.storedData_[this.surveyName_];
+    if (surveyData) {
+      Object.keys(surveyData).forEach(key => {
+        const id = this.normalizeIdAttr_(surveyData[key]);
+        const inp = this.querySelector(`#${id}`);
+        if (inp) inp.checked = true;
```
Comment (skarEE) [inline google-codelab-survey/google_codelab_survey.js]: {}'s

File: google-codelab-survey/google_codelab_survey.js (hunk 10 above)
Comment (arkassay) [reply google-codelab-survey/google_codelab_survey.js]: Done.

File: google-codelab-survey/google_codelab_survey.js (hunk 9 above)
Comment (arkassay) [reply google-codelab-survey/google_codelab_survey.js]: Done.

File: google-codelab-survey/google_codelab_survey.js (hunk 12 above)
Comment (arkassay) [reply google-codelab-survey/google_codelab_survey.js]: Done.

**Pull request merged**